### ISSUE TITLE
when caffe version release?

### ISSUE THREAD

**Issue by 10183308** [ZhouYanzhao/SPN#2]

**Comment by yeezhu** [comment]
@10183308 
A 3rd-party Caffe implementation developed by @KevinKecc is under testing. 
It would be released as soon as possible. Thanks for your patience!

**Comment by mychina75** [comment]
Waiting for the caffe solution too...

**Comment by zimenglan-sysu-512** [comment]
hi @yeezhu ,
do you still have a plan to release the caffe version?
thanks.

**Comment by yeezhu** [comment]
@zimenglan-sysu-512 
The Caffe code will not be released in the near future for some personal reasons.
Torch and PyTorch version are strongly recommended.

**Comment by zimenglan-sysu-512** [comment]
@yeezhu ,
Thanks for your reply. BTW, in training stage, the SPN does not need bp? do i understand it right?

**Comment by ZhouYanzhao** [comment]
@zimenglan-sysu-512 
SPNs are trained using standard classification settings, e.g., SGD & cross-entropy loss. Feel free to ping me (yzhou.work at outlook.com) if you would like to discuss or need further help.
